lib/client/Client.js

```diff
@@ -221,6 +222,96 @@ export default class Client {
   }
 
   /**
+   * Sends a _setMailboxes_ JMAP request.
+   *
+   * @param [options=null] {Object} The options to the _setMailboxes_ JMAP request.
+   *
+   * @return {Promise} A {@link Promise} that eventually resolves to a {@link Mailbox} object.
+   *
+   * @see http://jmap.io/spec.html#setmailboxes
+   */
+  setMailboxes(options) {
+    return this._jmapRequest('setMailboxes', options);
+  }
+
+  /**
+   * Creates a mailbox  by sending a _setMailboxes_ JMAP request.<br />
+   *
+   * @param name {String} The name of the mailbox to create.
+   * @param parentId {String} The id of the parent of the mailbox to create.
+   *
+   * @return {Promise} A {@link Promise} that eventually resolves to a {@link MailboxesSet}.
```

Comment:
Wrong, you're resolving with a `CreatedMailboxAck`, a new Class that you need to add. See `CreatedMessageAck` for a similar use case.